content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -138,6 +162,53 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+### dvc.lock file
+
+The purpose of `dvc.lock` is to lock the exact contents of all files involved in
+the pipeline(s) at the time that `dvc.yaml` was last updated.
```

Comment:
This does not sound like correct to me ... e.g.  I updated the `dvc.yaml` manually?

Also, `dvc.lock` also locks some other fields - params, not only files

Reply:
> This does not sound like correct to me ... e.g. I updated the dvc.yaml manually?

Good point. Rephrased to "at the time that the pipeline was last committed
(after using `dvc add`, `dvc run`, `dvc repro`, or `dvc commit`, for example)" in 2560ffed. Better?

> dvc.lock also locks some other fields - params, not only files

Yeah actually params files are not locked by md5. Only the param key/value pairs. This was already explained in my previous commit.

Reply:
> Yeah actually params files are not locked by md5. 

they are locked by values, values play the same role as md5s

> "at the time that the pipeline was last committed

no, sounds still confusing ... I think the whole approach should be different? For every stage in the pipeline it captures specific versions of deps, outs, etc involved. It serves a few purposes - detect changes (e.g. on repro, or status), the same as `.dvc` files for outputs and intermediate pipeline results - for commands like checkout to work, for get and import to work, etc

`dvc add` probably does not affect it, btw.

Reply:
> they are locked by values, values play the same role as md5s

Not exactly, you can add any contents to a params file, change indentation or spacing, maybe add comments, etc. and nothing will change from for `dvc status`. That's why I'm saying the params file itself is not locked to (tracked).

Anyway, will review this again ⏳

Reply:
> it captures specific versions of deps, outs, etc

Idk if mentioning "versions" here is best. DVC doesn't actually provide versioning: you create versions by 1. tracking stuff with DVC and 2. committing the meta files to Git. That's the way I've been seeing it recently (since I began writing the new Get Started).

But I'll review this text along these lines:

> It serves a few purposes - detect changes (e.g. on repro, or status), the same as .dvc files for outputs and intermediate pipeline results - for commands like checkout to work, for get and import to work, etc

as well as trying to incorporate Saugat's suggestion from https://github.com/iterative/dvc.org/pull/1448#discussion_r443125758 (on the value of dvc.lock).

Reply:
> detect changes (e.g. on repro, or status), the same as .dvc files for outputs

`.dvc` files help detect changes? Only for import stages which have dependencies, I think.

Anyway, I just updated this whole intro to dvc.lock in d7500f4a. PTAL